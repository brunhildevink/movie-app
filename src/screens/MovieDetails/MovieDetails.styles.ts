import styled from 'styled-components'
import { typography } from '../../styles'

const EpisodeInformation = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 40px;
  border-bottom: 1px solid rgba(151, 151, 151, 0.2);
`

const Image = styled.div<{ img: string }>`
  background: ${({ img }) => `url(${img})`} center center no-repeat;
  background-size: cover;
  height: 100%;
  width: 100%;
`

const Rating = styled.div`
  display: flex;
  align-items: center;

  svg {
    margin-right: 16px;
  }

  span {
    font-weight: ${typography.fontWeightBold};
  }
`

const TitleInformation = styled.div`
  padding: 40px;
  overflow-y: auto;
`

const Wrapper = styled.div`
  height: 100%;
  display: grid;
  grid-template-rows: 60% 15% 25%;
  grid-template-columns: 1fr;
`

export { EpisodeInformation, Image, Rating, TitleInformation, Wrapper }
