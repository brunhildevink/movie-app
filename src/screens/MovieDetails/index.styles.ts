import styled from 'styled-components'
import { breakpoints, colors, typography } from '../../styles'

const EpisodeInformation = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 40px;
  border-bottom: 1px solid rgba(151, 151, 151, 0.2);
`

const Image = styled.div<{ img: string }>`
  background: ${({ img }) => `url(${img})`} center center no-repeat;
  background-size: cover;
  width: 100%;
  height: 200px;
  grid-area: image;

  @media ${breakpoints.lg} {
    grid-area: unset;
    height: 100%;
  }
`

const NotFound = styled.div`
  background: ${colors.white};
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;

  svg {
    opacity: 0.5;
  }
`

const Rating = styled.div`
  display: flex;
  align-items: center;
  grid-area: rating;

  svg {
    margin-right: 16px;
  }

  span {
    font-weight: ${typography.fontWeightBold};
  }
`

const TitleInformation = styled.div`
  padding: 40px;
  overflow-y: auto;
  grid-area: title;

  p {
    font-size: ${typography.fontSizeTextLarge}px;
    line-height: ${typography.lineHeightTextLarge};
  }
`

const Wrapper = styled.div`
  height: 100%;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-areas:
    'rating'
    'title'
    'image';

  @media ${breakpoints.lg} {
    grid-template-areas:
      'image'
      'rating'
      'title';

    grid-template-rows: 60% 15% 25%;
  }
`

export { EpisodeInformation, Image, NotFound, Rating, TitleInformation, Wrapper }
