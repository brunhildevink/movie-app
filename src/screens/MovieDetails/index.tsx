import React from 'react'
import { EpisodeInformation, Image, Rating, TitleInformation, Wrapper } from './MovieDetails.styles'
import image from '../../assets/images/1.1.jpg'
import { Heading, Icon, Text } from '../../components'
import { colors } from '../../styles'

// interface Props {
//   date: string
//   description: string
//   imageIndex: number
//   rating: string
//   title: string
// }
const description =
  'Aute occaecat consequat labore occaecat do fugiat consectetur labore eu. Velit culpa amet labore voluptate ullamco proident sunt excepteur exercitation non incididunt eu esse labore. Culpa excepteur nulla irure ex minim cupidatat et voluptate esse Lorem ullamco laboris incididunt magna. Velit aute quis qui aliquip ullamco ad irure duis excepteur cillum excepteur officia nisi. Pariatur occaecat dolor anim enim. Sunt ex ullamco aute amet cupidatat ad proident cillum aliquip amet id velit.'
const title = 'The Mandalorian'
const episode = 1
const date = '20/20/2022'
const rating = 9

const MovieDetails: React.FC = () => (
  <Wrapper>
    <Image img={image} />
    <EpisodeInformation>
      <Heading.HeadingFour>
        Episode {episode} â€” {date}
      </Heading.HeadingFour>
      <Rating>
        <Icon.Star display="inline" fill={colors.yellow} height={24} title="rating" width={24} />
        <Heading.HeadingFour>
          <span>{rating}</span>/10
        </Heading.HeadingFour>
      </Rating>
    </EpisodeInformation>
    <TitleInformation>
      <Heading.HeadingThree>{title}</Heading.HeadingThree>
      <Text.Regular>{description}</Text.Regular>
    </TitleInformation>
  </Wrapper>
)

export default MovieDetails
