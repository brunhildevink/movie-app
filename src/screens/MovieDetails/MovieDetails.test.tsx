import React from 'react'
import renderer from 'react-test-renderer'
import MovieDetails from '.'

test('Header renders correctly', () => {
  const tree = renderer.create(<MovieDetails />).toJSON()
  expect(tree).toMatchSnapshot()
})
