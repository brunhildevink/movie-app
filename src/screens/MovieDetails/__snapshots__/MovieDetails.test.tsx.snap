// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails renders correctly 1`] = `
<div
  className="sc-eCYdqJ fBtlEz"
>
  <div
    className="sc-gsnTZi DWwtJ"
  />
  <div
    className="sc-bczRLJ cRUVyz"
  >
    <h4
      className="sc-ftvSup fYTEFs"
    >
      Episode 
      1
       â€” 
      20/20/2022
    </h4>
    <div
      className="sc-dkzDqf dXySZZ"
    >
      <svg
        style={
          Object {
            "display": "inline",
            "fill": "#efd358",
            "height": 24,
            "lineHeight": 1,
            "verticalAlign": "middle",
            "width": 24,
          }
        }
        viewBox="0 0 28 28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          rating
        </title>
        <path
          d="m14 .692 4.324 8.761 9.668 1.406-6.996 6.819 1.651 9.63L14 22.76l-8.647 4.547 1.65-9.63-6.995-6.82 9.668-1.405L14 .693Z"
          fill="#efd358"
        />
      </svg>
      <h4
        className="sc-ftvSup fYTEFs"
      >
        <span>
          9
        </span>
        /10
      </h4>
    </div>
  </div>
  <div
    className="sc-hKMtZM daYKig"
  >
    <h3
      className="sc-iBkjds huGiaA"
    >
      The Mandalorian
    </h3>
    <p
      className="sc-iqcoie kYFXBb"
    >
      Aute occaecat consequat labore occaecat do fugiat consectetur labore eu. Velit culpa amet labore voluptate ullamco proident sunt excepteur exercitation non incididunt eu esse labore. Culpa excepteur nulla irure ex minim cupidatat et voluptate esse Lorem ullamco laboris incididunt magna. Velit aute quis qui aliquip ullamco ad irure duis excepteur cillum excepteur officia nisi. Pariatur occaecat dolor anim enim. Sunt ex ullamco aute amet cupidatat ad proident cillum aliquip amet id velit.
    </p>
  </div>
</div>
`;
